internal/database/repository/node.go

```diff
@@ -57,48 +59,168 @@ func applyNodeFilters(builder *sql.Builder, filters NodeFilters) {
+			"attributes":      node.Attributes,
+			"capacity":        node.Capacity,
+			"allocated":       node.Allocated,
+			"occupied":        node.Occupied,
+			"available":       node.Available,
+			"utilized":        node.Utilized,
+			"allocations":     node.Allocations,
+			"schedulable":     node.Schedulable,
+			"is_reserved":     node.IsReserved,
+			"reservations":    node.Reservations,
+		})
+	if err != nil {
+		return fmt.Errorf("could not insert node into DB: %v", err)
 	}
 	return nil
 }
 
+func (s *PostgresRepository) UpdateNode(ctx context.Context, node *model.Node) error {
+	const q = `
+UPDATE nodes
+SET
+	deleted_at_nano = @deleted_at_nano,
+	node_id = @node_id,
+	partition = @partition,
+	host_name = @host_name,
+	rack_name = @rack_name,
+	attributes = @attributes,
+	capacity = @capacity,
+	allocated = @allocated,
+	occupied = @occupied,
+	available = @available,
+	utilized = @utilized,
+	allocations = @allocations,
+	schedulable = @schedulable,
+	is_reserved = @is_reserved,
+	reservations = @reservations
+WHERE id = @id`
+
+	res, err := s.dbpool.Exec(
+		ctx,
+		q,
+		pgx.NamedArgs{
+			"id":              node.ID,
+			"deleted_at_nano": node.DeletedAtNano,
+			"node_id":         node.NodeID,
+			"partition":       node.Partition,
+			"host_name":       node.HostName,
+			"rack_name":       node.RackName,
+			"attributes":      node.Attributes,
+			"capacity":        node.Capacity,
+			"allocated":       node.Allocated,
+			"occupied":        node.Occupied,
+			"available":       node.Available,
+			"utilized":        node.Utilized,
+			"allocations":     node.Allocations,
+			"schedulable":     node.Schedulable,
+			"is_reserved":     node.IsReserved,
+			"reservations":    node.Reservations,
+		})
+	if err != nil {
+		return fmt.Errorf("could not update node in DB: %v", err)
+	}
+	if res.RowsAffected() == 0 {
+		return fmt.Errorf("failed to update node %q: no rows affected", node.ID)
+	}
+
+	return nil
+}
+
+func (s *PostgresRepository) GetNodeByID(ctx context.Context, id string) (*model.Node, error) {
+	const q = ` SELECT * FROM nodes WHERE id = $1 ORDER BY id DESC LIMIT 1`
+	var node model.Node
+	row := s.dbpool.QueryRow(ctx, q, id)
+	if err := row.Scan(
+		&node.ID,
+		&node.CreatedAtNano,
+		&node.DeletedAtNano,
+		&node.NodeID,
+		&node.Partition,
+		&node.HostName,
+		&node.RackName,
+		&node.Attributes,
+		&node.Capacity,
+		&node.Allocated,
+		&node.Occupied,
+		&node.Available,
+		&node.Utilized,
+		&node.Allocations,
+		&node.Schedulable,
+		&node.IsReserved,
+		&node.Reservations,
+	); err != nil {
+		return nil, fmt.Errorf("could not get node from DB: %v", err)
+	}
+	return &node, nil
+}
+
+func (s *PostgresRepository) DeleteNodesNotInIDs(ctx context.Context, ids []string, deletedAtNano int64) error {
+	const q = `
+UPDATE nodes 
+SET deleted_at_nano = @deleted_at_nano
+WHERE deleted_at_nano IS NULL AND NOT (id = ANY(@ids))`
+	_, err := s.dbpool.Exec(
+		ctx,
+		q,
+		pgx.NamedArgs{
+			"ids":             ids,
```

Comment:
Can you swap the order of these two lines, so they match the order of the params in the SQL? It would be a little bit easier to follow.

Reply:
updated